### DISCUSSION TITLE
build to dynamic link lib (expose function to other program)

### DISCUSSION THREAD

**Discussion by mo-han** [pyinstaller/pyinstaller#6908] [Features & Development]
yeah i understand pyinstaller is designed to build binary executable, but what if it could also build into DLL (or wrapper DLL)? then it can help exposing python functions to other programs.

**Comment by bwoodsend** [comment]
That is a very very long way away from what PyInstaller does. And Python functions are not converted into ELF/PE/macho symbols - they're compiled to pycache (which is a Python only format) then added as data files to the executable. If you did go through the process of recompiling PyInstaller's bootloaders into shared libraries, built an application with it then inspected what symbols it had given you then you would only find some useless archive unpacking functions. Python's C API functions remain in `libpython.so` and the Python code (including some of the code to load it) is hidden as a data section.
